File: swift/Sources/WikimediaMetricsPlatform/Curation/CurationController.swift
import Foundation

class CurationController {
    let contextController: ContextController

    init(contextController: ContextController) {
        self.contextController = contextController
    }

    func eventPassesCurationRules(_ event: Event, streamConfig: StreamConfig) -> Bool {
        guard let curationRules = streamConfig.getCurationRules() else {
            return true
        }
        
        return curationRules.apply(to: event)
    }
}

File: java/src/main/java/org/wikimedia/metrics_platform/curation/CurationController.java
package org.wikimedia.metrics_platform.curation;

import javax.annotation.ParametersAreNonnullByDefault;
import javax.annotation.concurrent.ThreadSafe;

import org.wikimedia.metrics_platform.Event;
import org.wikimedia.metrics_platform.StreamConfig;

@ThreadSafe
@ParametersAreNonnullByDefault
public class CurationController {

    public boolean eventPassesCurationRules(Event event, StreamConfig streamConfig) {
        StreamConfig.ProducerConfig producerConfig = streamConfig.getProducerConfig();
        if (producerConfig == null) {
            return true;
        }
        StreamConfig.MetricsPlatformClientConfig metricsPlatformClientConfig =
                producerConfig.getMetricsPlatformClientConfig();
        if (metricsPlatformClientConfig == null) {
            return true;
        }
        CurationFilter curationFilter = metricsPlatformClientConfig.getCurationFilter();
        if (curationFilter == null) {
            return true;
        }
        return curationFilter.apply(event);
    }

}
